src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatResponse.cs

```diff
@@ -3,58 +3,100 @@
+    /// <param name="message">The response message.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="message"/> is <see langword="null"/>.</exception>
     public ChatResponse(ChatMessage message)
     {
         _ = Throw.IfNull(message);
-        _choices = [message];
+        Messages.Add(message);
     }
 
-    /// <summary>Gets or sets the list of chat response choices.</summary>
-    public IList<ChatMessage> Choices
+    /// <summary>Initializes a new instance of the <see cref="ChatResponse"/> class.</summary>
+    /// <param name="messages">The response messages.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="messages"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="messages"/> must not be read-only.</exception>
+    public ChatResponse(IList<ChatMessage> messages)
     {
-        get => _choices;
-        set => _choices = Throw.IfNull(value);
+        _ = Throw.IfNull(messages);
+        _ = Throw.IfReadOnly(messages);
+
+        _messages = messages;
     }
 
-    /// <summary>Gets the chat response message.</summary>
+    /// <summary>Gets or sets the chat response messages.</summary>
     /// <remarks>
-    /// If there are multiple choices, this property returns the first choice.
-    /// If <see cref="Choices"/> is empty, this property will throw. Use <see cref="Choices"/> to access all choices directly.
+    /// The last message in the list maps to <see cref="Message"/>. It should represent
+    /// the final result message of the operation.
+    /// </remarks>
+    /// <exception cref="ArgumentException">The <see cref="IList{T}"/> must not be read-only.</exception>
+    public IList<ChatMessage> Messages
+    {
+        get => _messages ??= new List<ChatMessage>(1);
+        set
+        {
+            if (value is not null)
+            {
+                _ = Throw.IfReadOnly(value);
+            }
+
+            _messages = value;
+        }
+    }
+
+    /// <summary>Gets or sets the last chat response message.</summary>
+    /// <remarks>
+    /// When getting <see cref="Message"/>, if there are no messages, <see cref="Message"/> will add a new
+    /// empty message to the list and return that message; if there are messages, the last will be returned.
+    /// When setting <see cref="Message"/>, if there are messages, the last message will be replaced by the
+    /// newly set instance; if there are no messages, the newly set instance will be added to the list.
     /// </remarks>
     [JsonIgnore]
     public ChatMessage Message
     {
         get
         {
-            var choices = Choices;
-            if (choices.Count == 0)
+            if (Messages.Count == 0)
             {
-                throw new InvalidOperationException($"The {nameof(ChatResponse)} instance does not contain any {nameof(ChatMessage)} choices.");
+                Messages.Add(new ChatMessage(ChatRole.Assistant, []));
```

Comment:
If there aren't messages why are we adding a non-existing message assuming it is a `Assistant` with no content. Maybe consider `Message` to be nullable and return null if `Count == 0`.

Reply:
@stephentoub This was marked as resolved, what's the rational to not returning null if there's no messages?

Wasn't clear why `null` is not an option where `throw` was acceptable before.

Reply:
> This was marked as resolved

Because this property no longer exists.